### PR TITLE
Kwxm/remove nop builtins (SCP-2267)

### PR DESCRIPTION
**Pull request by kwxm**
This moves the Nop builtins from the main set into a thing of their own in the benchmarking code, which is the only place they're needed.

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

File: plutus-core/cost-model/budgeting-bench/Bench.hs [hunk 1]
```diff
@@ -4,59 +4,74 @@
...
+type PlainTerm fun = UPLC.Term Name DefaultUni fun ()
 
 -- TODO.  I'm not totally sure what's going on here.  `env` is supposed to
 -- produce data that will be supplied to the things being benchmarked.  Here
 -- we've got a term and we evaluate it to get back the budget consumed, but then
 -- we throw that away and evaluate the term again.  This may have the effect of
 -- avoiding warmup, which could be a good thing.  Let's look into that.
-runTermBench :: String -> PlainTerm -> Benchmark
-runTermBench name term = env
+-- benchWith :: Int -> String -> PlainTerm fun -> Benchmark
```
Comment (michaelpj) [inline plutus-core/cost-model/budgeting-bench/Bench.hs:39]: commented out

**Review by michaelpj — APPROVED** [review]

File: plutus-core/cost-model/budgeting-bench/Bench.hs (hunk 1 above)
Comment (kwxm) [reply plutus-core/cost-model/budgeting-bench/Bench.hs:39]: Well spotted: it got swallowed up by the adjoining comment.

**Review by effectfully — COMMENTED** [review]
Only nitpicking.

File: plutus-core/cost-model/budgeting-bench/Nops.hs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+   units instead of integers, but using integers makes it possible to check that
+   the cost of calling the functions doesn't depend on the size of the
+   arguments.  We have checked this and there there was no dependence: let's
+   leave open the possibility of doing it again in case anything changes.
+-}
+instance (uni `Contains` Integer, GEq uni, GShow uni) => ToBuiltinMeaning uni NopFuns where
+    type CostingPart uni NopFuns = NopCostModel
+    toBuiltinMeaning
+        :: HasConstantIn uni term
+           => NopFuns -> BuiltinMeaning term NopCostModel
```
[the remark this answers is not in the document]
Comment (kwxm) [reply plutus-core/cost-model/budgeting-bench/Nops.hs:80]: You're right.  I'll try to remember to remove that some time.

**Pull request merged**